sockets/1.2/modules/ROOT/pages/sockets-documentation.adoc

```diff
@@ -594,7 +594,9 @@ Define your own custom protocol by writing a class that extends `TcpProtocol`.
 [[eof-protocol]]
 === Eof Protocol
 
-Reading is terminated by the stream being closed by the client.
+When clients connect to a server using the EOF protocol, they write the message on the socket connection output stream. To signal the end of the message, clients must signal the end of file (EOF) by closing the output stream and keeping the input stream open, awaiting for a response from the server. This behavior is known as half close. Then, the server sends a response and closes its output stream and the whole socket connection.
+
+For TLS secured connections, use TLS 1.3 and JDK 11. Half-Close feature is not supported in TLS 1.2 and earlier versions. The half-close feature triggers the close notification alert in order to prevent TLS truncation attacks. JDK11 and newer versions support TLS 1.3.
```

Comment:
```suggestion
For TLS secured connections, use TLS 1.3 and JDK 11 as the half-close feature is not supported in TLS 1.2 and earlier versions. The half-close feature triggers the close notification alert to prevent TLS truncation attacks. JDK 11 and newer versions support TLS 1.3.
```